src/inference/hmc.jl

```diff
@@ -594,5 +671,6 @@ function HMCState(
     # Unlink everything.
     invlink!(vi, spl)
 
-    return HMCState(vi, 0, 0, kernel.τ, h, AHMCAdaptor(spl.alg, metric; ϵ=ϵ), t.z)
+    adaptor = as_concrete(spl.alg.adaptor, metric; n_adapts=spl.alg.n_adapts, ϵ=ϵ, δ=spl.alg.δ)
+    return HMCState(vi, 0, 0, kernel.τ, h, spl.alg.adaptor, t.z)
```

Comment:
I guess this should be
```suggestion
    return HMCState(vi, 0, 0, kernel.τ, h, adaptor, t.z)
```